qutebrowser/mainwindow/tabwidget.py

```diff
@@ -439,11 +445,24 @@ def tabSizeHint(self, index):
             # get scroll buttons as soon as needed.
             size = minimum_size
         else:
+            #TODO: relative size and/or configured one
+            try:
+                tab = objreg.get('tab', scope='tab', window=self._win_id, tab=index)
```

Comment:
I took a look on the indicator-color, and I think I'm missing how it could help here. Maybe you can throw some light on this :)
I need two information: If the current tab is pinned or not, and how many pinned tabs are currently.
The first one to set the tab size, and the second of so we can calculate the size of the tabs that are not pinned.